src/core/ext/filters/client_channel/resolver/dns/c_ares/grpc_ares_wrapper.cc

```diff
@@ -144,12 +144,12 @@ static void grpc_ares_request_unref_locked(grpc_ares_request* r) {
 void grpc_ares_complete_request_locked(grpc_ares_request* r) {
   /* Invoke on_done callback and destroy the
      request */
+  r->ev_driver = nullptr;
```

Comment:
I assume that the ev_driver will clean itself up?  We're not holding a ref to it here?

We really ought to go ahead and combine the ev_driver and ares_request objects into one.  I don't think there's any good reason for them to be separate.

Reply:
Yes this is actually invoked from the driver object's dtor, so I figured it made sense to null the request objects pointer to it.

But yes, I think combining the request and driver objects into one will simplify things. I think it should be fairly straight forward so I'll add that change in

Reply:
Super.  It would probably be better to do that as a separate PR.

Reply:
sounds good, will do a separate PR for that